api/apps/api/src/modules/published-project/published-project.service.ts

```diff
@@ -59,13 +64,41 @@ export class PublishedProjectService {
       return left(alreadyPublished);
     }
 
-    // @debt scenarioId will be used in the png_map_data generation.
-    const { scenarioId, ...projectWithoutScenario } = projectToPublish;
+    // @debt If we end up moving the scenario map thumbnail generation
+    // to the end of the Marxan run process, this part here regarding
+    // the webshot should be removed and/or adapted to it. It looks
+    // like it does not belong here at all anyways, but right now there
+    // is not a better place to deal with this.
+
+    const webshotUrl = AppConfig.get('webshot.url') as string;
+
+    const { scenarioId, config, ...projectWithoutScenario } = projectToPublish;
+    assertDefined(scenarioId);
+    assertDefined(config);
+    const pngData = await this.webshotService.getPublishedProjectsImage(
+      scenarioId,
+      id,
+      {
+        ...config,
+        screenshotOptions: {
+          clip: { x: 0, y: 0, width: 500, height: 500 },
+        },
+      },
+      webshotUrl,
+    );
+
+    if (isLeft(pngData)) {
+      console.info(
+        `Map screenshot for public project ${id} could not be generated`,
+      );
+    }
```

Comment:
this would be a good candidate for breaking out into a new method (not sure we could reuse much of it when moving this step to the marxan run workflow)... feel free to leave it as is though, especially as we may move this soon in any case.

Reply:
I think for sure what I can reuse is the config stuff, as it is the same for BLM than this images. Apart from that maybe a common method in webshotService lib to distinguish between blm and publishedProject images? I don't know...

Reply:
i'd say let's leave it as is in this PR, so that we can merge and go ahead with the full flow between frontend and backend, and if/when we move this to marxan run flow, make it a self-contained method (the marxan run flow is complicated enough already that keeping new steps kind of self-contained would be quite useful)